zebra-state/src/service/finalized_state/disk_format.rs

```diff
@@ -253,6 +327,14 @@ impl DiskSerialize for rocksdb::WriteBatch {
         let value_bytes = value.as_bytes();
         self.put_cf(cf, key_bytes, value_bytes);
     }
+
+    fn zs_delete<K>(&mut self, cf: &rocksdb::ColumnFamily, key: K)
+    where
+        K: IntoDisk + Debug,
+    {
+        let key_bytes = key.as_bytes();
+        self.delete_cf(cf, key_bytes);
+    }
```

Comment:
Nice, I also wanted to make this method.

Do we want to assert `zs_contains` before deleting the key, and make that part of the `zs_delete` contract?

After this merges, I'd also like to replace the existing UTXO delete with `zs_delete`.
(If we delete a note commitment tree or UTXO that isn't there, Zebra should panic, because that's a bug.)

Reply:
You're right, it does get a bit complicated. And the non-finalized state already does the UTXO check.

I've added the `zs_delete` change to #2518, we don't have to do it in this PR.